integration-tests/bot/join_test.py

```diff
@@ -0,0 +1,331 @@
+#!/usr/bin/python3
+import docker
+import json
+import requests
+import datetime
+import time
+import jsonrpcclient
+from chainrpc import RPC, Blockchain
+from decouple import config
+from chainbot import SigningKey
+CURRENT_HASH = config('CURRENT_HASH', '')
+class Program :
+    def __init__(self) :
+        self.rpc = RPC()
+        self.blockchain = Blockchain()
+        # wallet a
+        self.node0_address = ""
+        self.node0_address1 = ""
+        self.node0_transfer_address = ""
+        self.node0_mnemonics= ""
+
+        # wallet b
+        self.node1_address = ""
+        self.node1_address1= ""
+        self.node1_transfer_address = ""
+        self.node1_mnemonics=""
+
+        # wallet b
+        self.node2_address = ""
+        self.node2_mnemonics=""
+
+
+
+        self.headers = {
+            'Content-Type': 'application/json',
+        }
+
+    def get_containers(self) :
+        client = docker.from_env()
+        containers= client.containers.list()
+        ret= {}
+        for container in containers:
+            id = container
+            #ret[id.name]= id.id
+            ret[id.name]= container
+        return ret
+        
+
+    #show_containers()
+    # tendermint rpc
+
+
+
+    def get_staking_state(self,name, passphrase, addr):
+        return self.rpc.staking.state(addr, name)
+       
+
+    def create_staking_address(self,name, passphrase):
+        return self.rpc.address.create(name,'staking')
```

Comment:
It needs to pass around encryption key(or called authentication token) now, or it probably won't run correctly.

Reply:
yes, i modified using enckey